packages/back-end/src/scim/users/patchUser.ts

```diff
@@ -0,0 +1,136 @@
+  userIndex: number,
+  user: UserInterface,
+  updatedScimUser: ScimUserObject
+) {
+  const updatedOrg = cloneDeep(org);
+
+  // When we introduce the ability to manage roles via SCIM, we can remove this check.
+  const userIsAdmin = org.members[userIndex].role === "admin";
+
+  if (userIsAdmin) {
+    const numberOfAdmins = org.members.filter(
+      (member) => member.role === "admin"
+    );
+
+    if (numberOfAdmins.length === 1) {
+      throw new Error("Cannot remove the only admin");
+    }
+  }
+
+  updatedOrg.members.splice(userIndex, 1);
+
+  await updateOrganization(org.id, updatedOrg);
+  await removeExternalId(user.id);
+
+  updatedScimUser.active = false;
+}
+
+export async function patchUser(req: ScimUpdateRequest, res: Response) {
+  // Get all of the params and operations
+  const requestBody = req.body;
+  const org = req.organization;
+
+  // Check if the user exists at all
+  // After this is all said and done, we need to return the user object
+  const user = await getUserByExternalId(req.params.id);
+  if (!user) {
+    return res.status(404).json({
+      schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+      status: "404",
+      detail: "User not found",
+    });
+  }
+
+  // Check if the user exists within the org
+  const userIndex = org.members.findIndex((member) => member.id === user.id);
+
+  // Then, we need to loop through operations
+  const operations: Operation[] = requestBody.Operations;
+
+  const updatedScimUser = {
+    schemas: ["urn:ietf:params:scim:schemas:core:2.0:User"],
+    id: req.params.id,
+    userName: user.email,
+    name: {
+      displayName: user.name || "",
+      givenName: user.name?.split(" ")[0],
+      familyName: user.name?.split(" ")[1],
+    },
+    active: userIndex > -1, // If a user has an externalId but is not in the org they're inactive
+    emails: [
+      {
+        primary: true,
+        value: user.email,
+        type: "work",
+        display: user.email,
+      },
+    ],
+    groups: [],
+    meta: {
+      resourceType: "User",
+    },
+  };
+  for (const operation of operations) {
+    const { op, value } = operation;
+    // Okta will only ever use PATCH to active/deactivate a user or sync a user's password
+    // https://developer.okta.com/docs/reference/scim/scim-20/#update-a-specific-user-patch
+    if (op === "replace" && value.active === false) {
+      // SCIM determines whether a user is active or not based on this property. If set to false, that means they want us to remove the user
+      // this means they want us to remove the user
+      try {
+        await removeUserFromOrg(org, userIndex, user, updatedScimUser);
+      } catch (e) {
+        return res.status(400).json({
+          schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+          status: "400",
+          detail: "Cannot remove the only admin",
+        });
+      }
+    } else if (op === "replace" && value.active === true) {
+      // TODO: Add user to org
```

Comment:
With our current setup, I'm not sure this will ever happen.

 Currently, when they're deactivated, they're removed from the org and when they're trying to be reactivated Okta does the following:
   - makes a get request to /Users?filter=userName eq "userName" - and with that request, we get all members of the org that have an externalId. The problem is, the user we're trying to reactivate isn't a member of the org anymore (this is because okta thinks we'll just change their "active" property to false when we deactivate them. But instead, we remove them entirely.)
  - So the listRequest returns an empty list response, so Okta makes a call the createUser.
  (It's my understanding that IF the listResponse returned a valid user, Okta would then make a call to patchUser with 
  
  In the createUser endpoint we have logic that looks to see if the user exists by email address or by externalId and if so, rather than creating a new user, we just update the existing user (if their email or name has changed) and add them to the organization, essentially "reactivating" them.
  
  FWIW - I don't love this twisty logic as it complicates the `createUser` method by doing more than just creating a user.
  
  So, we either keep it like it is, or we need to change the logic in the listUser endpoint to return users who WERE in the org, but were removed (aka, deactivated). I just don't know if that is feasible as we only have their userName (which is their email in the GB DB).

Reply:
With the new `managedByIdp` boolean I think we can do something where we grab all users that are managed by an idp and determine whether or not they're active depending on if they're in the orgs members array. The function we have currently to determine orphaned users might work for this

Reply:
@msamper From what I recall, the concept of an orphaned user is exclusive to self-hosted. If we implemented that to the cloud, it seems like we would be leaking user data for users outside of a particular organization.

E.G. If multiple orgs are using SCIM, they'll all have users that have the `managedByIdp` boolean, and we have no way of knowing which users belong to which org, so if we return all users that aren't in the org's member array that have the managedByIdp boolean, that would include users who were created for other organizations.